sonar-orchestrator/src/main/java/com/sonar/orchestrator/locator/GitHubImpl.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Orchestrator
+ * Copyright (C) 2011-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package com.sonar.orchestrator.locator;
+
+import com.eclipsesource.json.Json;
+import com.eclipsesource.json.JsonValue;
+import com.sonar.orchestrator.http.HttpCall;
+import com.sonar.orchestrator.http.HttpClientFactory;
+import okhttp3.HttpUrl;
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.Optional;
+
+public class GitHubImpl implements GitHub {
+  private static final Logger LOG = LoggerFactory.getLogger(GitHubImpl.class);
+  private static String cachedVersion;
```

Comment:
Shouldn't this field be `volatile`?
You made the new methods `getLatestScannerReleaseVersion` and `resetCache` synchronized, from which I assume this class should be thread-safe.
Without the `volatile`, a test like `getLatestScannerReleaseVersion_two_calls_make_a_single_request`, but done on multiple threads, would become flaky, since `cachedVersion != null` would become local to the thread.